### PR TITLE
Add Youtube/Twitch sign-up page

### LINKED ISSUE #2843: Allow users to include Twitch channels in sign up
We added RSS-Bridge to our system, and it lets you create feeds for various services that don't support it.  For example, Twitch channels.

You can access our instance at https://dev.api.telescope.cdot.systems/v1/rss-bridge/

If a user signs up for Telescope and has a Twitch channel, we should let them put in their Twitch channel info, then construct a URL to use our RSS-Bridge instance so we can add it as a feed.

I don't use Twitch so I don't have a good example here.  Maybe @sirinoks can point us at a channel we can use as an example.

cc @dbelokon, @sirinoks

### PR DESCRIPTION
**Pull request by menghif**
## Issue This PR Addresses

Fixes #2843 - Work in progress

## Type of Change

- [ ] **Bugfix**: Change which fixes an issue
- [x] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Description

I originally planned with @tcvan0707 to add a Youtube and Twitch field under the Blog RSS page but it ended up looking crowded and difficult to understand.

This PR adds a new page in the signup flow for the 2 optional Youtube and Twitch fields.

Screen Shot 2022-04-19 at 11 54 16 AM

## Steps to test the PR

- `pnpm services:start`
- `cd src/db`
- `cp env.example .env`
- `pnpm migrate`

## Checklist

- [x] **Quality**: This PR builds and passes our npm test and works locally
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

File: src/web/app/src/components/SignUp/Forms/YoutubeTwitchFeeds.tsx [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+    <div className={classes.root}>
+      <div className={classes.container}>
+        <h1 className={classes.blogPageTitle}>Youtube and Twitch</h1>
+        <h2 className={classes.helpText}>
+          OPTIONAL: Enter your Twitch URL and select the RSS you want to use in Telescope ecosystem.
+        </h2>
+        <div className={classes.infoContainer}>
+          <div className={classes.inputsContainer}>
+            <TextInput
+              name={twitchUrl.name}
```
[the remark this answers is not in the document]
Comment (menghif) [reply src/web/app/src/components/SignUp/Forms/YoutubeTwitchFeeds.tsx]: Good idea, I changed everything to use the word `channel`.

File: src/web/app/src/components/SignUp/Forms/ChannelFeeds.tsx [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+        <h2 className={classes.helpText}>
+          OPTIONAL: Enter your YouTube and/or Twitch channels and select the RSS feed(s) you want to
+          use in Telescope (separate more than one URL with a space)
+        </h2>
+        <div className={classes.infoContainer}>
+          <div className={classes.inputsContainer}>
+            <TextInput
+              name={channelUrl.name}
+              label={channelUrl.label}
+              helperText={channelUrlError || 'Validate your Youtube or Twitch URL(s)'}
```
Comment (humphd) [inline src/web/app/src/components/SignUp/Forms/ChannelFeeds.tsx:231]: YouTube (capital 'T')

File: src/web/app/src/components/SignUp/Schema/FormModel.tsx [hunk 3]
```diff
@@ -35,10 +35,15 @@ export default {
...
   blogUrl: {
     name: 'blogUrl',
-    label: 'Blog URl',
+    label: 'Blog URL',
     requiredErrorMsg: 'Blog URL is required',
     invalidErrorMsg: 'Invalid URL',
   },
+  channelUrl: {
+    name: 'channelUrl',
+    label: 'Youtube/Twitch URL(s)',
```
Comment (humphd) [inline src/web/app/src/components/SignUp/Schema/FormModel.tsx:44]: YouTube

**Review by humphd — CHANGES_REQUESTED** [review]

**Comment by tcvan0707** [comment]
I am thinking we should include the YouTube/Twitch links on the Review page too

**Comment by menghif** [comment]
> I am thinking we should include the YouTube/Twitch links on the Review page too

They are included.

**Review by tcvan0707 — APPROVED** [review]

**Pull request merged**